mobile/test/swift/EngineBuilderTests.swift

```diff
@@ -408,6 +392,42 @@ final class EngineBuilderTests: XCTestCase {
     self.waitForExpectations(timeout: 0.01)
   }
 
+  func testAddingRtdsAndAdsConfigurationWhenRunningEnvoy() {
+    let expectation = self.expectation(description: "Run called with expected data")
+    MockEnvoyEngine.onRunWithConfig = { config, _ in
+      XCTAssertTrue(config.generateYamlString().contains("rtds_layer_name"))
```

Comment:
These are now the only tests for swift or objective C which actually assert that the generated config is right - the current tests just assert that the EnvoyConfiguration object has stored the input params correctly which is trivial.  I'd be inclined to do a follow-up PR to move all tests here over to the new style.

Reply:
+1, this is a good test and we could use more of them!